src/test/java/org/ohdsi/analysis/versioning/SemverUtilsTest.java

```diff
@@ -0,0 +1,98 @@
+package org.ohdsi.analysis.versioning;
+
+import com.vdurmont.semver4j.SemverException;
+import org.junit.Test;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.ohdsi.analysis.versioning.SemverUtils.getRangesIntersection;
+
+public class SemverUtilsTest {
+
+
+    @Test
+    public void testGetRangesIntersectionSingle() {
+        String intersection = getRangesIntersection(singletonList(">=5.2"));
+        assertEquals(">=5.2", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionSimple() {
+        String intersection = getRangesIntersection(asList(">=5.0.0", ">=3.0.0"));
+        assertEquals(">=5.0.0", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionNoIntersection() {
+        String intersection = getRangesIntersection(asList(">=5.0.0", "3.0.0"));
+        assertNull(intersection);
+    }
+    
+    @Test
+    public void testGetRangesIntersectionWithSame1() {
+        String intersection = getRangesIntersection(asList(">=5.0.0", "5.0.0"));
+        assertEquals("5.0.0", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionWithSame2() {
+        String intersection = getRangesIntersection(asList(">=5.0.0", ">5.0.0"));
+        assertEquals(">5.0.0", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionWithSame3() {
+        String intersection = getRangesIntersection(asList("<=5.0.0", "<5.0.0"));
+        assertEquals("<5.0.0", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionWithSame4() {
+        String intersection = getRangesIntersection(asList(">=5.0.0", "<=5.0.0"));
+        assertEquals("5.0.0", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionWithRange1() {
+        String intersection = getRangesIntersection(asList(">=3.0.0", "<=5.0.0"));
+        assertEquals("3.0.0 - 5.0.0", intersection);
+    }
+
+    @Test
+    public void testGetRangesIntersectionWithRange2() {
+        String intersection = getRangesIntersection(asList("4 - 5", "<=5.0.0"));
+        assertEquals("4.0.0 - 5.0.0", intersection);
+    }
```

Comment:
Can you add a test that gets the intersection of `("4 - 5", "<=5")`.   I don't think it can resolve in a range of 4.0.0 - 5.0.0 because 5.0 is not included and I wanted to understand how that will resolve.

Reply:
You are right... it doesn't work, let me think about it a minute

Reply:
I think the hacky workaround is probably better than a 'correct' but complex (buggy) implementation, I've made a change and added a test